### ISSUE TITLE
Reproducing results on AGRA

### ISSUE THREAD

**Issue by altndrr** [HCPLab-SYSU/CD-FER-Benchmark#18]
Good morning,

I'm trying to reproduce the results of AGRA, but I cannot match the numbers. I believe the issue is due to hyperparameter selection and data processing. Can you confirm that the hyperparameters in the TrainOnSourceDomain.sh and TransferToTargetDomain.sh files are correct?

Also, can you help me understand the data processing step you performed? I'm using the RAF-DB dataset as source and JAFFE and FER2013 as targets. From what I read in #17, I believe you split the datasets into three equal-sized parts when no train-test split is available. If you have the test split, you divide it into two equal-sized test and validation sets.

Therefore, you use 1/3 of JAFFE and 1/2 of the test set of FER2013. For the RAF-DB, on the other hand, you directly use the train split provided by the authors.

Can you confirm these processes? 
Many thanks

**Comment by putao537** [comment]
I can't be sure if the hyperparameters are correct because it's been a long time since finishing these so many experiments.

As do you mention to split of dataset, I use the standard dataset split if the authors provided (e.g., RAF, FER2013), and otherwise randomly generate new dataset split (e.g., ExpW). However, there are too few images in JAFFE dataset, thus we set the all images of JAFFE as the train set without annotations.

**Comment by putao537** [comment]
> Good morning,
> 
> I'm trying to reproduce the results of AGRA, but I cannot match the numbers. I believe the issue is due to hyperparameter selection and data processing. Can you confirm that the hyperparameters in the TrainOnSourceDomain.sh and TransferToTargetDomain.sh files are correct?
> 
> Also, can you help me understand the data processing step you performed? I'm using the RAF-DB dataset as source and JAFFE and FER2013 as targets. From what I read in #17, I believe you split the datasets into three equal-sized parts when no train-test split is available. If you have the test split, you divide it into two equal-sized test and validation sets.
> 
> Therefore, you use 1/3 of JAFFE and 1/2 of the test set of FER2013. For the RAF-DB, on the other hand, you directly use the train split provided by the authors.
> 
> Can you confirm these processes? Many thanks

I can't be sure if the hyperparameters are correct because it's been a long time since finishing these so many experiments.

As do you mention to split of dataset, I use the standard dataset split if the authors provided (e.g., RAF, FER2013), and otherwise randomly generate new dataset split (e.g., ExpW). However, there are too few images in JAFFE dataset, thus we set the all images of JAFFE as the train set without annotations.

**Comment by altndrr** [comment]
Then, can I ask you the weights of the backbones? I was not able to extract them from the pre-trained models and load them into the architectures.

**Comment by putao537** [comment]
> Then, can I ask you the weights of the backbones? I was not able to extract them from the pre-trained models and load them into the architectures.

To extract weights from the pre-trained models, you may need run getPreTrainedModel_ResNet.py (or getPreTrainedModel_MobileNet.py)

**Comment by altndrr** [comment]
But I would like to extract the weights only for the backbones, without GCN and the other components you introduce. For the three architectures you test (ResNet18, ResNet50 and MobileNetv2), I'd like to get only their weights to plug inside vanilla architectures.

**Comment by putao537** [comment]
> But I would like to extract the weights only for the backbones, without GCN and the other components you introduce. For the three architectures you test (ResNet18, ResNet50 and MobileNetv2), I'd like to get only their weights to plug inside vanilla architectures.

Set all parameters to False. For instance, to get ResNet-50, modify the parameters in the line 6 in getPreTrainedModel_ResNet.py.
